### PR TITLE
[BEAM-119] Integrate DisplayData into DataflowPipelineRunner

### PR DESCRIPTION
**Pull request by swegner**
Translate registered display data into the job steps used by DataflowPipelineRunner.

### CODE REVIEW

**Comment by swegner** [comment]
R: @bjchambers @kennknowles

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java [hunk 1]
```diff
@@ -725,6 +729,16 @@ private void addOutput(String name, PValue value, Coder<?> valueCoder) {
       outputInfoList.add(outputInfo);
     }
 
+    private void addDisplayData(String name, DisplayData displayData) {
+      List<Map<String, Object>> serializedItems = Lists.newArrayList();
+      ObjectMapper mapper = MAPPER.setSerializationInclusion(JsonInclude.Include.NON_NULL);
```
Comment (bjchambers) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java]: Does this mutate MAPPER? 
If yes, isn't that a problem?
If no, would it make sense to have a separate static "NON_NULL_MAPPER" so we don't need to create a new mapper for every step? Any performance risk there?

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayData.java [hunk 2]
```diff
@@ -267,7 +267,7 @@ public String getLabel() {
      * <p>If no URL was specified, this will return {@code null}.
      */
     @Nullable
-    public String getUrl() {
+    public String getLinkUrl() {
```
Comment (bjchambers) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayData.java]: May be worth using an explicit `@JsonGetter("link_url")` on this and the other getters to ensure stability of the JSON names (and make it clear that these getters are being used by Jackson).

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslatorTest.java [hunk 3]
```diff
@@ -796,4 +801,63 @@ public void testToIterableTranslationWithIsmSideInput() throws Exception {
...
+                .add("foo", "bar")
+                .add("foo2", DataflowPipelineTranslatorTest.class)
+                .withLabel("Test Class")
+                .withLinkUrl("http://www.google.com");
+      }
+    };
+
+    pipeline
+      .apply(Create.of(1, 2, 3))
+      .apply(ParDo.of(fn));
```
Comment (bjchambers) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslatorTest.java]: May be worthwhile to have a test with two annotated steps, and make sure that (1) they both get the appropriate annotations and (2) they don't accidentally get annotations from the other.

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java [hunk 4]
```diff
@@ -548,6 +551,7 @@ public void addStep(PTransform<?, ?> transform, String type) {
       currentStep.setKind(type);
       steps.add(currentStep);
       addInput(PropertyNames.USER_NAME, getFullName(transform));
+      addDisplayData(PropertyNames.DISPLAY_METADATA, DisplayData.from(transform));
```
Comment (bjchambers) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java]: I'd propose we call the field DISPLAY_DATA (and put it in a similarly named location in the JSON). This way everything lines up.

**Comment by bjchambers** [comment]
LGTM.

@kennknowles in your court

**Comment by kennknowles** [comment]
LGTM

**Pull request merged**